### DISCUSSION TITLE
LLM Based Evaluation - more control with mapping a variable e.g. {{query}}, {{ground_truth}} etc inside of Input/Output/Metadata observation

### DISCUSSION THREAD

**Discussion by stepanogil** [langfuse/langfuse#3352] [Ideas]
### Describe the feature or potential improvement

This is how you currently map variables in the model-based Evaluation (Beta) module:

![image](https://github.com/user-attachments/assets/6b4bb862-279a-4b04-befe-27dde85adb8c)

i.e. you map whole returns of Input or Output or Metadata to a variable e.g. {{query}}

This is an example of how I populate the a Trace Input:

![image](https://github.com/user-attachments/assets/2731ad93-d703-431d-b958-c06a65891b1f)

This means when I do Evals, this whole json blob is passed on as the value of my {{query}}:

![image](https://github.com/user-attachments/assets/836965f6-41ef-4eee-8149-d0a7a2a20eb9)

Which I don't want to do as all the metadata will just introduce a lot of noise to the eval.
What I want to do is map my {{query}} to particular kv pair inside of the Input. Example below:

![image](https://github.com/user-attachments/assets/f2727a46-b91b-4f53-9122-6bb9a5350a75)

So I want another dropdown here where I can configure to which key I want to map the {{query}} variable.
So I can input something like: `Input["conversation_history"][0]["content"]` or something to this effect.

![image](https://github.com/user-attachments/assets/62256c3b-e7e6-4e91-b913-9b3cf8121225)

Hope that makes sense!

Also pls support Azure OpenAI in Evaluation - should be easy to implement- it has the same function calling implementation as OpenAI which seems to be integral to Evaluation.
Thats it. more power! im a big langfuse fan!

**Comment by marcklingen** [comment]
Great feedback, thanks for taking the time to share this in this great detail!

AzureOpenAI was added in the meantime and can now be used for evaluation in Langfuse.

Being able to query (e.g. json-path) within an input/output object is tracked here already: https://github.com/orgs/langfuse/discussions/3237

**Comment by maxdeichmann** [comment]
@stepanogil we built and released this yesterday. Please let us know if this works for you.

https://github.com/orgs/langfuse/discussions/3237
